spring-kafka/src/main/java/org/springframework/kafka/requestreply/ReplyingKafkaTemplate.java

```diff
@@ -100,6 +107,50 @@ public void setReplyTimeout(long replyTimeout) {
 		this.replyTimeout = replyTimeout;
 	}
 
+	@Override
+	public boolean isRunning() {
+		return this.running;
+	}
+
+	@Override
+	public int getPhase() {
+		return this.phase;
+	}
+
+	public void setPhase(int phase) {
+		this.phase = phase;
+	}
+
+	@Override
+	public boolean isAutoStartup() {
+		return this.autoStartup;
+	}
+
+	public void setAutoStartup(boolean autoStartup) {
+		this.autoStartup = autoStartup;
+	}
+
+	/**
+	 * Return the topics configured for the replying listener container.
+	 * @param patternConsumer called if the container is subscribed to a pattern.
+	 * @return the topics or an empty list if the container is subscribed to a pattern.
+	 */
+	public List<String> obtainReplyTopics(Consumer<Pattern> patternConsumer) {
```

Comment:
I don't feel like this `Consumer` is better than just return a set of configured topics/patters and let end-user to filter or process them as it is convenient for them.

OTOH why can't we return a set of **assigned** topics from the:
```
/**
 * Return the {@link TopicPartition}s currently assigned to this container,
 * either explicitly or by Kafka; may be null if not assigned yet.
 * @return the {@link TopicPartition}s currently assigned to this container,
 * either explicitly or by Kafka; may be null if not assigned yet.
 */
public Collection<TopicPartition> getAssignedPartitions() {
```
?

Seems for me this way we don't need to worry about patterns at all.

WDYT?

Reply:
AH - good catch; thanks; forgot about that.